pkg/mesh/topology.go

```diff
@@ -314,51 +336,58 @@ func (t *Topology) Conf() *wireguard.Conf {
 
 // AsPeer generates the WireGuard peer configuration for the local location of the given Topology.
 // This configuration can be used to configure this location as a peer of another WireGuard interface.
-func (t *Topology) AsPeer() *wireguard.Peer {
+func (t *Topology) AsPeer() wireguard.Peer {
 	for _, s := range t.segments {
 		if s.location != t.location {
 			continue
 		}
-		return &wireguard.Peer{
-			AllowedIPs: s.allowedIPs,
-			Endpoint:   s.endpoint,
-			PublicKey:  s.key,
+		p := wireguard.Peer{
+			PeerConfig: wgtypes.PeerConfig{
+				AllowedIPs: s.allowedIPs,
+				PublicKey:  s.key,
+			},
+			Endpoint: s.endpoint,
 		}
+		return p
 	}
-	return nil
+	return wireguard.Peer{}
 }
 
 // PeerConf generates a WireGuard configuration file for a given peer in a Topology.
-func (t *Topology) PeerConf(name string) *wireguard.Conf {
-	var pka int
-	var psk []byte
+func (t *Topology) PeerConf(name string) wireguard.Conf {
```

Comment:
Should this return a pointer again?

Reply:
I think so. What was the reason for changing it?

Reply:
I don't know. I will add a check to `(*Conf).Bytes()` to return `nil, nil` in case `*Conf == nil`.